### PR TITLE
Restructure `Model` for future flexibility

### PR DESCRIPTION
**Pull request by michplunkett**
## Describe your changes
Restructure the `Model` class with flexibility in mind, make the model isolatable for Jupyter notebooks, and install Jupyter into the project.

## Checklist before requesting a review
- [x] The code runs successfully.

```
(poetry-python-project-py3.11) michaelp@MacBook-Air-3 supreme-court-ml-predictions % make logistic-regression
python -m "supreme_court_predictions" --logistic-regression
oh, what up?
------------------------------------------
Running regression on total_utterances...
Creating bag of words
Starting the Logistic Regression
Accuracy score: 0.5405405405405406
------------------------------------------
------------------------------------------
Running regression on judge_utterances...
Creating bag of words
Starting the Logistic Regression
Accuracy score: 0.5205479452054794
------------------------------------------
------------------------------------------
Running regression on advocate_utterances...
Creating bag of words
Starting the Logistic Regression
Accuracy score: 0.7567567567567568
------------------------------------------
------------------------------------------
Running regression on adversary_utterances...
Creating bag of words
Starting the Logistic Regression
Accuracy score: 0.7837837837837838
------------------------------------------
(poetry-python-project-py3.11) michaelp@MacBook-Air-3 supreme-court-ml-predictions % 
```
- [x] I ran `make lint` and have made the bulk of changes that it has requested.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -20,6 +20,7 @@ scikit-learn = "^1.2.2"
 spacy = "^3.5.2"
 ruff = "^0.0.263"
 xgboost = "^1.7.5"
+jupyter = "^1.0.0"
```
Comment (michplunkett) [inline pyproject.toml:23]: Add `jupyter` to the project.

File: second_checkpoint.ipynb [hunk 2]
```diff
@@ -5,7 +5,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## Exploratory Analysis - Tests"
+    "## Second Checkpoint Essay"
```
Comment (michplunkett) [inline second_checkpoint.ipynb:8]: The next PR will be heavily modifying this file.

File: supreme_court_predictions/models/logistic_regression.py [hunk 3]
```diff
@@ -88,7 +88,7 @@ def run(self):
             try:
                 print("------------------------------------------")
                 print(f"Running regression on {df_name}...")
-                acc = self.create(df)
+                acc = self.create_and_measure(df)
```
Comment (michplunkett) [inline supreme_court_predictions/models/logistic_regression.py:91]: This function is located in the `Model` class.

File: supreme_court_predictions/models/xg_boost.py [hunk 4]
```diff
@@ -101,7 +100,7 @@ def run(self):
             try:
                 print("------------------------------------------")
                 print(f"Running a gradient boosted tree model on {df_name}...")
-                acc = self.create(df)
+                acc = self.create_and_measure(df, accuracy_score)
```
Comment (michplunkett) [inline supreme_court_predictions/models/xg_boost.py:103]: You pass in the accuracy function you intend to use.

**Review by FedericoDM — APPROVED** [review]

**Review by chanteriam — APPROVED** [review]

**Pull request merged**